ffmpeg/encoder.c

```diff
@@ -376,6 +379,20 @@ int mux(AVPacket *pkt, AVRational tb, struct output_ctx *octx, AVStream *ost)
   if (AVMEDIA_TYPE_AUDIO == ost->codecpar->codec_type) {
       if (octx->drop_ts == AV_NOPTS_VALUE) octx->drop_ts = pkt->pts;
       if (pkt->pts && pkt->pts == octx->drop_ts) return 0;
+
+      if (pkt->dts != AV_NOPTS_VALUE && pkt->pts != AV_NOPTS_VALUE && pkt->dts > pkt->pts) {
+        pkt->pts = pkt->dts = pkt->pts + pkt->dts + octx->last_audio_dts + 1
+                     - FFMIN3(pkt->pts, pkt->dts, octx->last_audio_dts + 1)
+                     - FFMAX3(pkt->pts, pkt->dts, octx->last_audio_dts + 1);
+      }
+      if (pkt->dts != AV_NOPTS_VALUE && octx->last_audio_dts != AV_NOPTS_VALUE) {
+        int64_t max = octx->last_audio_dts + !(octx->oc->oformat->flags & AVFMT_TS_NONSTRICT);
```

Comment:
Looks like `last_audio_dts` used before initialization here? That's a bit esoteric snippet, maybe add a comment on what it does?

Reply:
Added detail comments for last_audio_dts variable, and this pipeline is ported from [here](https://github.com/livepeer/FFmpeg/blob/682c4189d8364867bcc49f9749e04b27dc37cded/fftools/ffmpeg.c#L824)